lib/LazyLoadingGhostFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager;
+
+use ProxyManager\Configuration;
+use ProxyManager\Factory\LazyLoadingGhostFactory as BaseLazyLoadingGhostFactory;
+
+/**
+ * Extends the LazyLoadingGhostFactory of the ProxyManager, to make it configureable according to the needs of Sulu.
+ */
+class LazyLoadingGhostFactory extends BaseLazyLoadingGhostFactory
+{
+    public function __construct($cacheDirectory)
+    {
+        if (!is_dir($cacheDirectory)) {
+            mkdir($cacheDirectory);
```

Comment:
Note that there was a recent security venrability in Doctrine whereby cache files created with the permissions 777 could be modified by any user on the system to execute any code as the web user. We should probably review Sulu for this security problem generally.